Quantized LSTM/GRU: Remove legacy API support

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #72522

Ref #72263 for cpp_custom_type_hack removal

These overloads were deprecated in #35787 which was in the PyTorch 1.6
release, so the BC period is well expired.

cc @jamesr66a

Differential Revision: [D34111271](https://our.internmc.facebook.com/intern/diff/D34111271)

peterbell10 committed:
Quantized LSTM/GRU: Remove legacy API support

Ref #72263 for cpp_custom_type_hack removal

These overloads were deprecated in #35787 which was in the PyTorch 1.6
release, so the BC period is well expired.

[ghstack-poisoned]

peterbell10 committed:
Update on "Quantized LSTM/GRU: Remove legacy API support"


Ref #72263 for cpp_custom_type_hack removal

These overloads were deprecated in #35787 which was in the PyTorch 1.6
release, so the BC period is well expired.

[ghstack-poisoned]

albanD reviewed (approved):
LGTM

cc @jerryzh168 does that look good on your end as well?

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34111271).

peterbell10 committed:
Rebase on "Quantized LSTM/GRU: Remove legacy API support"


Ref #72263 for cpp_custom_type_hack removal

These overloads were deprecated in #35787 which was in the PyTorch 1.6
release, so the BC period is well expired.

cc jamesr66a

Differential Revision: [D34111271](https://our.internmc.facebook.com/intern/diff/D34111271)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34111271).
